Compact documentation is added. [API-1937] (#1141)
hazelcast/hazelcast-cpp-client: Reference_Manual.md

Note that, this is very similar to `identified_data_serializer` implementation except the method signatures and the derived marker class of `portable_serializer` for the specialized serializer.

`hz_serializer<Person>` specialization of `hz_serializer` should implement the above four methods, namely `get_factory_id`, `get_class_id`, `write_portable`, `read_portable`. In case that the object fields are non-public, you can always define struct `hz_serializer<Person>` as friend to your object class. You use the `portable_writer` class methods when serializing the object into binary bytes and you use the `portable_reader` class methods when de-serializing the bytes into the concrete object instance.

> ## 4.3. compact_serializer Serialization

Reviewer: Can you please add a link to [this](https://docs.hazelcast.com/hazelcast/5.3-snapshot/compact-binary-specification) page for complete reference on binary details of the serialization.

Author: Fixed.

> As an enhancement to existing serialization methods, Hazelcast offers compact serialization, with the following main features.

Reviewer 2: ```suggestion
As an enhancement to the existing serialization methods, Hazelcast offers compact serialization, with the following main features:
```

Author: Fixed.

> - Separates the schema from the data and stores it per type, not per object which results in less memory and bandwidth usage compared to other formats

Reviewer 2: ```suggestion
- Separates the schema from the data and stores it for each type instead of each object which results in less memory and bandwidth usage compared to other formats,
```

Author: Fixed.

- Does not require a class to implement an interface or change the source code of the class in any way
- Supports schema evolution which permits adding or removing fields, or changing the types of fields

> - Can work any kind of for classes and structs

Reviewer 2: ```suggestion
- Can work with any kind of types including classes and structs.
```

Author: Fixed.

> - Can work with any kind of types

Reviewer: what does this mean?
